flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesClusterDescriptor.java

```diff
@@ -214,9 +215,12 @@ public ClusterClientProvider<String> deployApplicationCluster(
 
         applicationConfiguration.applyToConfiguration(flinkConfig);
 
-        // No need to do pipelineJars validation if it is a PyFlink job.
+        // No need to do pipelineJars validation if it is a PyFlink job or no jars specified in
+        // pipeline.jars (to use jars in system classpath instead).
         if (!(PackagedProgramUtils.isPython(applicationConfiguration.getApplicationClassName())
-                || PackagedProgramUtils.isPython(applicationConfiguration.getProgramArguments()))) {
+                        || PackagedProgramUtils.isPython(
+                                applicationConfiguration.getProgramArguments()))
+                && flinkConfig.get(PipelineOptions.JARS) != null) {
```

Comment:
Would it make be useful to push this compound condition to `PackageProgramUtils` as something like `hasClasspath` or `usesSystemClasspath` (with the implied flip to false)?

Reply:
good idea!